src/net/send_recv/msg.rs

```diff
@@ -591,6 +597,75 @@ impl<'buf> Iterator for AncillaryDrain<'buf> {
 
 impl FusedIterator for AncillaryDrain<'_> {}
 
+/// An ABI-compatible wrapper for `mmsghdr`, for sending multiple messages with
+/// [sendmmsg].
+#[cfg(target_os = "linux")]
+#[repr(transparent)]
+pub struct MMsgHdr<'a> {
+    raw: c::mmsghdr,
+    _phantom: PhantomData<&'a mut ()>,
+}
+
+#[cfg(target_os = "linux")]
+impl<'a> MMsgHdr<'a> {
+    /// Constructs a new message with no destination address.
+    pub fn new(iov: &[IoSlice<'a>], control: &mut SendAncillaryBuffer<'_, '_, '_>) -> Self {
```

Comment:
Just leaving a note for myself that I think the lifetimes are also being elided incorrectly here - because technically by using `Self` there's no elision on the output lifetime. So I think this works out to:

```rust
pub fn new(iov: &'x [IoSlice<'a>], control: &'y mut SendAncillaryBuffer<'_, '_, '_>) -> MMsgHdr<'a> {}
```